api/app/Actions/Event/GetEventCollectionAction.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Event;
+
+use App\Actions\PaginatedResponse;
+use App\Repositories\Event\Criterion\EventTypesCriterion;
+use App\Repositories\Event\EventRepository;
+use App\Repositories\Event\EventRepositoryInterface;
+use App\Repositories\Event\Criterion\StartDateCriterion;
+use App\Repositories\Event\Criterion\EndDateCriterion;
+use App\Repositories\Event\Criterion\OwnerCriterion;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Auth;
+
+final class GetEventCollectionAction
+{
+    private EventRepositoryInterface $repository;
+
+    public function __construct(EventRepositoryInterface $eventRepository)
+    {
+        $this->repository = $eventRepository;
+    }
+
+    public function execute(GetEventCollectionRequest $request): PaginatedResponse
+    {
+        $criteria = [new OwnerCriterion(Auth::id())];
+
+        if ($request->getStartDate()) {
+            $startDate = Carbon::parse($request->getStartDate())->format('Y-m-d');
```

Comment:
`start_date` and `end_date` should contain time too for showing correct results for past and upcoming events.